rwengine/src/engine/GameState.hpp

```diff
@@ -316,7 +316,9 @@ class GameState {
 
     bool skipCutscene = false;
     bool isIntroPlaying = false;
-    CutsceneData* currentCutscene = nullptr;
+
+    std::unique_ptr<CutsceneData> currentCutscene;
```

Comment:
nit: with cpp 17 we can avoid dynamic allocation when using optional.

Reply:
`optional` would indeed simplify a lot of things.
Boost also has an optional.